src/dynsight/_internal/utilities/utilities.py

```diff
@@ -16,3 +19,42 @@ def normalize_array(x: np.ndarray) -> np.ndarray:  # type: ignore[type-arg]
     norm = np.linalg.norm(x, axis=-1, keepdims=True)
     norm[norm == 0] = 1
     return x / norm
+
+
+def find_extrema_points(
+    x_axis: np.ndarray[float, Any],
+    y_axis: np.ndarray[float, Any],
+    extrema_type: str,
```

Comment:
use `extrema_type: Literal["min", "max"],` instead so that the user knows it is one or the other.

Reply:
`from typing import Literal`

Reply:
using this can i avoid this part ? 
```python
    if extrema_type not in {"min", "max"}:
        type_msg = "extrema_type must be 'min' or 'max'"
        raise ValueError(type_msg)
```

Reply:
`Literal` does not raise an error at run time, this is just for type checking, so you should keep this check